app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/minigame/FireOutbreakActivity.java

```diff
@@ -1,7 +1,185 @@
+import nl.tudelft.pixelperfect.pixelperfect.R;
+import nl.tudelft.pixelperfect.pixelperfect.Spaceship;
 
 /**
- * Cre
+ * Class responsible for the handling of the fire outbreak event.  A choice must
+ * be made between multiple solutions, and then selected where to
+ * deploy them to combat the fire.
+ *
+ * @author David Alderliesten
+ *
  */
-public class FireOutbreakActivity {
+public class FireActivity  extends AppCompatActivity {
+    private Spaceship ship;
+    private GameClient game;
+
+    private int fireLocation;
+
+    private CheckBox useFireExtinguisher;
+    private CheckBox useAutomator;
+    private CheckBox useWaterBucket;
+    private CheckBox useFireBlanket;
+    private CheckBox useHammer;
+
+    private Button deployToAirSupport;
+    private Button deployToEngine;
+    private Button deployToWingPlasma;
+
+    /**
+     * On creation of the fire activity the content views will be initialized
+     * and relevant information retrieved.
+     *
+     * @param savedInstanceState the instance to create.
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_mini_game_fire);
+
+        useFireExtinguisher = (CheckBox) findViewById(R.id.extinguishBox);
+        useAutomator = (CheckBox) findViewById(R.id.automatedKillerBox);
+        useWaterBucket = (CheckBox) findViewById(R.id.bucketBox);
+        useFireBlanket = (CheckBox) findViewById(R.id.blanketBox);
+        useHammer = (CheckBox) findViewById(R.id.hammerBox);
+
+        deployToAirSupport = (Button) findViewById(R.id.airSupportButton);
+        deployToEngine = (Button) findViewById(R.id.engineButton);
+        deployToWingPlasma = (Button) findViewById(R.id.wingPlasmaButton);
+
+        game = GameClient.getInstance();
+        ship = Spaceship.getInstance();
+    }
+
+    /**
+     * Class that validates the user action and returns the result to the server.
+     *
+     * @param passedState
+     *      The given state of validation.
+     */
+    private void validateAction(int passedState) {
+        if(!ship.getEventLog().contains(Events.FIRE)) {
+            game.sendMessage(new EventCompletedMessage("WRONG FIRE ACTIVATION MOMENT", -1));
+        }
+
+        if(passedState < 0) {
+            game.sendMessage(new EventCompletedMessage("WRONG FIRE COMBINATION", -1));
+        } else {
+            Event event = ship.getEventLog().peek(Events.FIRE);
+
+            game.sendMessage(new EventCompletedMessage("Asteroid Field Event", ship.getEventLog().pop(Events.FIRE).getId()));
+        }
+    }
+
+    /**
+     * Class aimed at verifying the total score obtained to verify the selected methods.
+     * A small value is good for the engines, a large value is good for the wings.
+     *
+     * @return total score found.
+     */
+    private int accumulatedVerification() {
+        // Int for verifying the accumulated selection.
```

Comment:
I would use the example for checkboxes on this page https://developer.android.com/guide/topics/ui/controls/checkbox.html . This helps clearing out the code a bit, since now you have too much private attributes.